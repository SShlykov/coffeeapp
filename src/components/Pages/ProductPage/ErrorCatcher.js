import React, {Component} from "react";
import ProductPage from "./ProductPage";
import ErrorMessage from "../../ErrorMessage";
import cDataService from "../../../services/cDataService";
import Spinner from "../../Item/ErrorCatcher";

export default class ErrorCatcher extends Component {
    state = {
        data: [],
        loader: true,
        error: false
    };
    _isMounted = false;
    cDataService = new cDataService();
    onItemLoaded = ( data ) => {
        if(this._isMounted) {
            this.setState({
                data,
                loading: false
            })
        }
    };
    onError = (error) => {
        if (this._isMounted) {
            this.setState({
                error: error,
                loading: false
            })
        }
    };
    loadItem = (id) => {
        this.cDataService.getCoffee(id)
            .then(data => {
                if(data.description){
                if(data.description.length>200){
                    data.smallDescr = `${data.description.slice(0,200)}...`
                    }}
                return data
            })
            .then(this.onItemLoaded)
            .catch((error)=> error.name||error.message)
            .then(this.onError)
    };
    componentDidMount() {
        this._isMounted = true;
        this.loadItem(this.props.id)
    }
    componentDidCatch(error) {
        this.setState({
            error: error.name||error.message
        })
    }
    componentWillUnmount() {
        this._isMounted = false;
    }

    render() {
        const { error, loading } = this.state;
        const content = this.state.loading? <Spinner/>:
            !(loading || error || !this.state.data )? <ProductPage {...this.state.data}/>:
                (!this.state.data)?<ErrorMessage error="We cant found product you searching" button/>: <ErrorMessage error={error} button/>;
        return(
            <>{content}</>
        )
    }
}