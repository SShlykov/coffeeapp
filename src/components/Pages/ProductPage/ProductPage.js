import React from "react";
import {BeansLogo} from "../../SmallComp";

const ProductPage = ({name, country, description, price, url}) => {
        return (
            <div className="productPage">
                <div className="container">
                    <div className="row">
                        <div className="col-lg-5 offset-1">
                            <img className="shop__girl" src={url} alt="coffee_item" style={{maxWidth: "100%"}}/>
                        </div>
                        <div className="col-lg-4">
                            <div className="title">About {name}</div>
                            <BeansLogo dark/>
                            <div className="shop__point">
                                <span>Country:</span>
                                {country}
                            </div>
                            <div className="shop__point">
                                <span>Description:</span>
                                {description}
                            </div>
                            <div className="shop__point">
                                <span>Price:</span>
                                <span className="shop__point-price">{price}</span>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
};
export default ProductPage;