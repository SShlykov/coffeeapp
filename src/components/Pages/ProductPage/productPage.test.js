import React from 'react';
import ProductPage from "./ProductPage";
import {shallow} from 'enzyme';

describe('Testing <ProductPage/>', () => {
    const item = shallow(<ProductPage/>);
    describe('Testing snap & state', () => {
        it('ProductPage have rendered correctly', () => {
            expect(item).toMatchSnapshot();
        });
    });
});