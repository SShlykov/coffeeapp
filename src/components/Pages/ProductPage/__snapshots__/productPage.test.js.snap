// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ProductPage/> Testing snap & state ProductPage have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-5 offset-1"
          >
            <img
              alt="coffee_item"
              className="shop__girl"
              style={
                Object {
                  "maxWidth": "100%",
                }
              }
            />
          </div>
          <div
            className="col-lg-4"
          >
            <div
              className="title"
            >
              About 
            </div>
            <BeansLogo
              dark={true}
            />
            <div
              className="shop__point"
            >
              <span>
                Country:
              </span>
            </div>
            <div
              className="shop__point"
              onClick={[Function]}
            >
              <span>
                Description:
              </span>
            </div>
            <div
              className="shop__point"
            >
              <span>
                Price:
              </span>
              <span
                className="shop__point-price"
              />
            </div>
          </div>
        </div>
      </div>,
      "className": "productPage",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-lg-5 offset-1"
          >
            <img
              alt="coffee_item"
              className="shop__girl"
              style={
                Object {
                  "maxWidth": "100%",
                }
              }
            />
          </div>
          <div
            className="col-lg-4"
          >
            <div
              className="title"
            >
              About 
            </div>
            <BeansLogo
              dark={true}
            />
            <div
              className="shop__point"
            >
              <span>
                Country:
              </span>
            </div>
            <div
              className="shop__point"
              onClick={[Function]}
            >
              <span>
                Description:
              </span>
            </div>
            <div
              className="shop__point"
            >
              <span>
                Price:
              </span>
              <span
                className="shop__point-price"
              />
            </div>
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-lg-5 offset-1"
            >
              <img
                alt="coffee_item"
                className="shop__girl"
                style={
                  Object {
                    "maxWidth": "100%",
                  }
                }
              />
            </div>,
            <div
              className="col-lg-4"
            >
              <div
                className="title"
              >
                About 
              </div>
              <BeansLogo
                dark={true}
              />
              <div
                className="shop__point"
              >
                <span>
                  Country:
                </span>
              </div>
              <div
                className="shop__point"
                onClick={[Function]}
              >
                <span>
                  Description:
                </span>
              </div>
              <div
                className="shop__point"
              >
                <span>
                  Price:
                </span>
                <span
                  className="shop__point-price"
                />
              </div>
            </div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="coffee_item"
                className="shop__girl"
                style={
                  Object {
                    "maxWidth": "100%",
                  }
                }
              />,
              "className": "col-lg-5 offset-1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "coffee_item",
                "className": "shop__girl",
                "src": undefined,
                "style": Object {
                  "maxWidth": "100%",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
                >
                  About 
                </div>,
                <BeansLogo
                  dark={true}
                />,
                <div
                  className="shop__point"
                >
                  <span>
                    Country:
                  </span>
                </div>,
                <div
                  className="shop__point"
                  onClick={[Function]}
                >
                  <span>
                    Description:
                  </span>
                </div>,
                <div
                  className="shop__point"
                >
                  <span>
                    Price:
                  </span>
                  <span
                    className="shop__point-price"
                  />
                </div>,
              ],
              "className": "col-lg-4",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "About ",
                    undefined,
                  ],
                  "className": "title",
                },
                "ref": null,
                "rendered": Array [
                  "About ",
                  undefined,
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "dark": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Country:
                    </span>,
                    undefined,
                  ],
                  "className": "shop__point",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Country:",
                    },
                    "ref": null,
                    "rendered": "Country:",
                    "type": "span",
                  },
                  undefined,
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Description:
                    </span>,
                    undefined,
                  ],
                  "className": "shop__point",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Description:",
                    },
                    "ref": null,
                    "rendered": "Description:",
                    "type": "span",
                  },
                  undefined,
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Price:
                    </span>,
                    <span
                      className="shop__point-price"
                    />,
                  ],
                  "className": "shop__point",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Price:",
                    },
                    "ref": null,
                    "rendered": "Price:",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "shop__point-price",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-lg-5 offset-1"
            >
              <img
                alt="coffee_item"
                className="shop__girl"
                style={
                  Object {
                    "maxWidth": "100%",
                  }
                }
              />
            </div>
            <div
              className="col-lg-4"
            >
              <div
                className="title"
              >
                About 
              </div>
              <BeansLogo
                dark={true}
              />
              <div
                className="shop__point"
              >
                <span>
                  Country:
                </span>
              </div>
              <div
                className="shop__point"
                onClick={[Function]}
              >
                <span>
                  Description:
                </span>
              </div>
              <div
                className="shop__point"
              >
                <span>
                  Price:
                </span>
                <span
                  className="shop__point-price"
                />
              </div>
            </div>
          </div>
        </div>,
        "className": "productPage",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col-lg-5 offset-1"
            >
              <img
                alt="coffee_item"
                className="shop__girl"
                style={
                  Object {
                    "maxWidth": "100%",
                  }
                }
              />
            </div>
            <div
              className="col-lg-4"
            >
              <div
                className="title"
              >
                About 
              </div>
              <BeansLogo
                dark={true}
              />
              <div
                className="shop__point"
              >
                <span>
                  Country:
                </span>
              </div>
              <div
                className="shop__point"
                onClick={[Function]}
              >
                <span>
                  Description:
                </span>
              </div>
              <div
                className="shop__point"
              >
                <span>
                  Price:
                </span>
                <span
                  className="shop__point-price"
                />
              </div>
            </div>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-lg-5 offset-1"
              >
                <img
                  alt="coffee_item"
                  className="shop__girl"
                  style={
                    Object {
                      "maxWidth": "100%",
                    }
                  }
                />
              </div>,
              <div
                className="col-lg-4"
              >
                <div
                  className="title"
                >
                  About 
                </div>
                <BeansLogo
                  dark={true}
                />
                <div
                  className="shop__point"
                >
                  <span>
                    Country:
                  </span>
                </div>
                <div
                  className="shop__point"
                  onClick={[Function]}
                >
                  <span>
                    Description:
                  </span>
                </div>
                <div
                  className="shop__point"
                >
                  <span>
                    Price:
                  </span>
                  <span
                    className="shop__point-price"
                  />
                </div>
              </div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="coffee_item"
                  className="shop__girl"
                  style={
                    Object {
                      "maxWidth": "100%",
                    }
                  }
                />,
                "className": "col-lg-5 offset-1",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "coffee_item",
                  "className": "shop__girl",
                  "src": undefined,
                  "style": Object {
                    "maxWidth": "100%",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="title"
                  >
                    About 
                  </div>,
                  <BeansLogo
                    dark={true}
                  />,
                  <div
                    className="shop__point"
                  >
                    <span>
                      Country:
                    </span>
                  </div>,
                  <div
                    className="shop__point"
                    onClick={[Function]}
                  >
                    <span>
                      Description:
                    </span>
                  </div>,
                  <div
                    className="shop__point"
                  >
                    <span>
                      Price:
                    </span>
                    <span
                      className="shop__point-price"
                    />
                  </div>,
                ],
                "className": "col-lg-4",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "About ",
                      undefined,
                    ],
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": Array [
                    "About ",
                    undefined,
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "dark": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        Country:
                      </span>,
                      undefined,
                    ],
                    "className": "shop__point",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Country:",
                      },
                      "ref": null,
                      "rendered": "Country:",
                      "type": "span",
                    },
                    undefined,
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        Description:
                      </span>,
                      undefined,
                    ],
                    "className": "shop__point",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Description:",
                      },
                      "ref": null,
                      "rendered": "Description:",
                      "type": "span",
                    },
                    undefined,
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        Price:
                      </span>,
                      <span
                        className="shop__point-price"
                      />,
                    ],
                    "className": "shop__point",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Price:",
                      },
                      "ref": null,
                      "rendered": "Price:",
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "className": "shop__point-price",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
